### PR TITLE
Add ember responsive types

### PR DESCRIPTION
**Pull request by dwilhelmi**

### CODE REVIEW

File: types/ember-animated/animated-container.d.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import Component from "@glint/environment-ember-loose/ember-component";
 
 interface AnimatedContainerSignature {
+  /** Multiple tags supported via `tag` arg, all should still be base HTMLElement. */
+  Element: HTMLElement;
```
Comment (wagenet) [inline types/ember-animated/animated-container.d.ts:5]:
```suggestion
  Element: Element;
```

SVGElements aren't HTMLElements

Comment (wagenet) [reply types/ember-animated/animated-container.d.ts:5]: Alternatively, we could do something fancy where we take a generic for the `tag` and then use Glint's `ElementForTagName` here.

**Review by wagenet — CHANGES_REQUESTED** [review]

File: types/ember-animated/animated-container.d.ts (hunk 1 above)
Comment (dwilhelmi) [reply types/ember-animated/animated-container.d.ts:5]: That sounds very promising, didn't know that was a thing! I can look into that.

Comment (dwilhelmi) [reply types/ember-animated/animated-container.d.ts]: Updated, that's a cool feature, I like it!

Comment (wagenet) [reply types/ember-animated/animated-container.d.ts]: @dwilhemli want to copy my tests from https://github.com/Gavant/glint-template-types/pull/28 ?

Comment (wagenet) [reply types/ember-animated/animated-container.d.ts]: I didn't expect you to get back to this so quickly!

Comment (dwilhelmi) [reply types/ember-animated/animated-container.d.ts]: done, though I'm getting some type errors locally, unsure if it is just an environment issue or a real problem

File: types/ember-animated/animated-container.d.ts [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
-    tag?: String
+    tag?: Tag
   };
   Yields: {
     default: [];
   };
 }
 
-export default class AnimatedContainer extends Component<AnimatedContainerSignature> {}
+declare class AnimatedContainer<Tag extends string = 'div'> extends Component<AnimatedContainerSignature<Tag>> {}
```
Comment (wagenet) [inline types/ember-animated/animated-container.d.ts:21]:
```suggestion
export default class AnimatedContainer<Tag extends string = 'div'> extends Component<AnimatedContainerSignature<Tag>> {}
```

**Review by wagenet — CHANGES_REQUESTED** [review]

File: types/ember-animated/animated-container.d.ts (hunk 2 above)
Comment (dwilhelmi) [reply types/ember-animated/animated-container.d.ts:21]: Doh! I am doing it as a declare in my local project I'm using these types in, accidentally copied that over

**Review by wagenet — APPROVED** [review]
Looks good to me. Should probably squash some commits before merging.

**Comment by dwilhelmi** [comment]
Rebase complete, commits squashed

**Pull request merged**